### PR TITLE
templates: Rework hooks.

### PR DESCRIPTION
**Pull request by Gacko**
Basically just an attempt to fix a weird issue...

This PR:

- Reworks the CRD install and cluster issuer hook. Also removes weights and delete policies.

#### Optional app

- [x] fresh install
- [x] upgrade from previous version

#### Pre-installed app

- [x] fresh install during cluster creation
- [x] upgrade from previous version in a pre-existing cluster

#### Other testing

- [x] check reconciliation of existing resources after upgrading

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by webwurst** [comment]
- I see `namespace: {{ .Release.Namespace }}` is dropped from several resources. Will those still be installed in the apps namespace by the chart-operator by default?
- Can you explain why the `helm.sh/hook-weight` didn't help us here?
- Is the main change the removal of `helm.sh/hook-delete-policy` from all the hooks?

**Comment by Gacko** [comment]
This PR is just an attempt of determining an issue with hooks / hook weights during cluster rollout on OpenStack. I therefore first of all need a test release in the `default-test` catalog and will later on optimize my changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Changed
+
+- Rework hooks.
```
Comment (ubergesundheit) [inline CHANGELOG.md:12]: IMO the changelog entry should be more descriptive towards what has changed and maybe why it was neccessary. In the sense of: What is the improvement introduced.

**Comment by Gacko** [comment]
@ubergesundheit Updated the changelog.

**Review by ubergesundheit — APPROVED** [review]
Changes look good to me and it looks like you certainly tested everything :+1: 

This feedback regarding the changelog entry is very picky and also very opinionated (and I certainly not always deliver on this myself all the time):

I think a changelog entries should be written in a way that the intended audience can quickly decide if its relevant for oneself or not. The audience may range between Giant Swarm employees and customers which a huge range of knowledge about either the app but also the templating done in the helm chart and involved other parts etc. This means IMO to include only relevant changes with maybe half of a sentence why the change was neccessary and what the impact is.

Meaning: This entry has too many details for my taste. It may be worth linking to this PR for people interested in the actual changes in the code.

Here is a suggestion:


> - Refactored the way CRDs and default ClusterIssuer are applied to resolve timeout problems [#263](https://github.com/giantswarm/cert-manager-app/pull/263)

**Comment by Gacko** [comment]
Hm, I think you're right and most details are not relevant for customers but may be for employees working on a customer request like "we would like to update `cert-manager`, what do we need to take care of?". I'd then simply review the changelog for them.

**Comment by Gacko** [comment]
I added a link to this PR. :)

**Pull request merged**